core/api/src/jsonrpc/impl/web3.rs

```diff
@@ -606,29 +606,32 @@ impl<Adapter: APIAdapter + 'static> Web3RpcServer for Web3RpcImpl<Adapter> {
 
     #[metrics_rpc("eth_getTransactionReceipt")]
     async fn get_transaction_receipt(&self, hash: H256) -> RpcResult<Option<Web3Receipt>> {
+        let ctx = Context::new();
         let res = self
             .adapter
-            .get_transaction_by_hash(Context::new(), hash)
+            .get_transaction_by_hash(ctx.clone(), hash)
             .await
             .map_err(|e| Error::Custom(e.to_string()))?;
 
         if let Some(stx) = res {
             if let Some(receipt) = self
                 .adapter
-                .get_receipt_by_tx_hash(Context::new(), hash)
+                .get_receipt_by_tx_hash(ctx.clone(), hash)
                 .await
                 .map_err(|e| Error::Custom(e.to_string()))?
             {
-                Ok(Some(Web3Receipt::new(receipt, stx)))
-            } else {
-                Err(Error::Custom(format!(
-                    "can not get receipt by hash {:?}",
-                    hash
-                )))
+                return Ok(Some(Web3Receipt::new(receipt, stx)));
             }
-        } else {
-            Ok(None)
         }
+
+        if self.adapter.mempool_contains_tx(ctx, &hash).await {
+            return Ok(None);
+        }
+
+        Err(Error::Custom(format!(
+            "Can not get receipt by hash {:?}",
+            hash
+        )))
```

Comment:
Suppose 0x85d995eba9763907fdf35cd2034144dd9d53ce32cbec21349d4b12823c6860c1 is a nonexistent transaction hash.
And the result from `eth_getTransactionReceipt` is null, not Error.
```bash
curl --location --request POST 'https://rpc.flashbots.net/' \
--header 'Content-Type: application/json' \
--header 'Accept: */*' \
--data-raw '{
    "id": 42,
    "jsonrpc": "2.0",
    "method": "eth_getTransactionReceipt",
    "params": ["0x85d995eba9763907fdf35cd2034144dd9d53ce32cbec21349d4b12823c6860c1"]
}'

# Result
{"id":42,"result":null,"jsonrpc":"2.0"}

```

**Note**: rpc.flashbots.net is [Ethereum Mainnet](https://chainlist.org/chain/1)'s RPC.

Reply:
Does this mean query any nonexistent transaction hash should return `Ok(None)`?

Reply:
I think it should returnA transaction receipt object, or `null` when no receipt was found.

See https://ethereum.org/en/developers/docs/apis/json-rpc/#eth_gettransactionreceipt